src/main/java/org/assertj/core/error/ShouldBeEqualWithinPercentage.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2015 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import org.assertj.core.data.Percentage;
+
+/**
+ * Creates an error message indicating that an assertion that
+ * verifies that two numbers are equal within a positive percentage failed.
+ *
+ * @author Alexander Bischof
+ */
+public class ShouldBeEqualWithinPercentage extends BasicErrorMessageFactory {
+
+    /**
+     * Creates a new <code>{@link org.assertj.core.error.ShouldBeEqualWithinPercentage}</code>.
+     *
+     * @param actual     the actual value in the failed assertion.
+     * @param expected   the expected value in the failed assertion.
+     * @param percentage the given percentage between 0 and 100.
+     * @param difference the effective difference between actual and expected.
+     * @return the created {@code ErrorMessageFactory}.
+     */
+    public static <T extends Number> ErrorMessageFactory shouldBeEqualWithinPercentage(T actual, T expected,
+                                                                                       Percentage<T> percentage,
+                                                                                       T difference) {
+        double expectedPercentage = difference.doubleValue() / expected.doubleValue() * 100d;
+        return new ShouldBeEqualWithinPercentage(actual, expected, percentage, expectedPercentage);
+    }
+
+    private <T extends Number> ShouldBeEqualWithinPercentage(Number actual, Number expected, Percentage<T> percentage,
+                                                             double expectedPercentage) {
+        super("%nExpecting:%n  <%s>%nto be close to:%n  <%s>%n" +
+              "by less than <%s> percent but difference was <%s> percent.%n" +
```

Comment:
better if the message looked like : `"by less than 10.0% but difference was 20.0% percent.`